### PR TITLE
Issue#607 sort part2

### PR DESCRIPTION
**Pull request by Maxstgt**
The second part of this issue. This PR includes:
- Removed unnecessary imports from app.module.ts (marketplace & dashboard).
- Sort by function for movies.
- List view for movies.
- Implemented akita-ng-fire in the mono repo and updated akita to v4, cause otherwise, akita-ng-fire won't work.
- The Movie store and Movie service are now using akita-ng-fire.
- New Guard which fetches all the movies in the collection. Also implements CollectionGuard form akita-ng-fire.
- Fixed some minor typos.
- Also renamed some functions in the movieService, cause akita-ng-fire now occupies the normal CRUD operations function names like `update`.

### CODE REVIEW

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts [hunk 1]
```diff
@@ -95,17 +97,78 @@ export class MarketplaceSearchComponent implements OnInit {
...
   ////////////////////
 
+  public sortBy(option?: string) {
+    switch (option) {
+      case 'Director':
+        this.movieSearchResults$
+          .pipe(
+            tap(movies => {
+              this.sortByOptionMovies = movies.sort((a: Movie, b: Movie) => {
+                if (b.main.directors[0].lastName < a.main.directors[0].lastName) {
```
Comment (phpgeekfr) [inline apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts]: you can replace this by `return b.main.directors[0].lastName.localeCompare(a.main.directors[0].lastName);`

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts [hunk 2]
```diff
@@ -95,17 +97,78 @@ export class MarketplaceSearchComponent implements OnInit {
...
+          .subscribe();
+        break;
+      case 'Title':
+        this.movieSearchResults$
+          .pipe(
+            tap(
+              movies =>
+                (this.sortByOptionMovies = movies.sort((a: Movie, b: Movie) => {
+                  if (a.main.title.original < b.main.title.original) {
+                    return -1;
```
Comment (phpgeekfr) [inline apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts]: same as above

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts [hunk 3]
```diff
@@ -95,17 +97,78 @@ export class MarketplaceSearchComponent implements OnInit {
...
+          )
+          .subscribe();
+        break;
+      case 'Production Year':
+        this.movieSearchResults$
+          .pipe(
+            tap(
+              movies =>
+                (this.sortByOptionMovies = movies.sort((a: Movie, b: Movie) => {
+                  if (b.main.productionYear < a.main.productionYear) {
```
Comment (phpgeekfr) [inline apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts]: same as above

File: apps/catalog/marketplace/marketplace/src/app/movie/view/view.component.ts [hunk 4]
```diff
@@ -9,7 +9,7 @@ import { MovieQuery } from '@blockframes/movie';
   styleUrls: ['./view.component.scss'],
   changeDetection: ChangeDetectionStrategy.Default
 })
-export class MovieViewComponent implements OnInit {
+export class CatalogMovieViewComponent implements OnInit {
```
[the remark this answers is not in the document]
Comment (Maxstgt) [reply apps/catalog/marketplace/marketplace/src/app/movie/view/view.component.ts]: yes, sounds better!

File: libs/movie/src/lib/movie/+state/movie.service.ts [hunk 5]
```diff
@@ -17,61 +17,66 @@ export function cleanModel<T>(data: T): T {
...
     // we don't want to keep orgId in our Movie object
     if (movie.organization) delete movie.organization;
     if (movie.stakeholders) delete movie.stakeholders;
 
-    return this.db.doc<Movie>(`movies/${id}`).update(cleanModel(movie));
-  }
+    return this.fireQuery.doc<Movie>(`movies/${id}`).update(cleanModel(movie));
+  } 
 
   public async remove(movieId: string): Promise<void> {
```
[the remark this answers is not in the document]
Comment (Maxstgt) [reply libs/movie/src/lib/movie/+state/movie.service.ts]: this doesn't interfere with the akita ng fire cause this function is async. So I didn't blow up the PR with even more file changes. But if you want, I can rename it?

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts (hunk 3 above)
Comment (Maxstgt) [reply apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts]: `localCompare` doesn't exist on number...? Any other ideas?

File: libs/movie/src/lib/movie/+state/movie.service.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/+state/movie.service.ts]: No it's ok, it was just for consistency

File: apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts (hunk 3 above)
Comment (phpgeekfr) [reply apps/catalog/marketplace/marketplace/src/app/movie/search/search.component.ts]: Ho yes sorry this work only for string. You could do `mynumber.toLocaleString().localeCompare()` but this is ugly ;). Keep your actual code for comparing number

**Pull request merged**